src/main/kotlin/com/ampnet/projectservice/service/impl/OrganizationInviteServiceImpl.kt

```diff
@@ -120,4 +126,25 @@ class OrganizationInviteServiceImpl(
             )
         }
     }
+
+    private fun throwExceptionForAlreadyInOrganization(organization: Organization, emails: List<String>) {
+        val users = userService.getUsersByEmail(emails)
+        if (users.isNotEmpty()) {
+            val memberships = organization.memberships
+            if (memberships?.isNotEmpty() == true) {
+                val userMembershipMap = memberships.associateBy { it.userUuid.toString() }
+                users.filter { user ->
+                    userMembershipMap.containsKey(user.uuid)
+                }
+            }
+            if (users.isNotEmpty()) {
+                val mails = users.joinToString { it.email }
+                throw ResourceAlreadyExistsException(
+                    ErrorCode.ORG_DUPLICATE_USER,
+                    "Some users are already a member of this organization: $mails",
+                    mapOf("emails" to mails)
+                )
+            }
+        }
```

Comment:
Looks like this logic is not correct. Function `filter` returns a new list with filtered elements [docs](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter.html). `users` list will never be empty.

Also, it is better to check the project-service database for memberships and fetch the user-service for users because it is cheaper operation.

```suggestion
        organization.memberships?.associateBy { it.userUuid.toString() }?.let { membershipMap ->
            val duplicatedUsers = userService.getUsersByEmail(emails)
                .filter { membershipMap.containsKey(it.uuid) }
            if (duplicatedUsers.isNotEmpty()) {
                val duplicatedEmails = duplicatedUsers.joinToString { it.email }
                throw ResourceAlreadyExistsException(
                    ErrorCode.ORG_DUPLICATE_USER,
                    "Some users are already a member of this organization: $duplicatedEmails",
                    mapOf("emails" to duplicatedEmails)
                )
            }
        }
```